src/CattyTests/DataModel/VariablesContainerTest.swift

```diff
@@ -234,6 +234,121 @@ final class VariablesContainerTest: XCTestCase {
+        objectB.name = "testObjectB"
+
+        let list = UserVariable(name: "testList", isList: true)
+
+        let container1 = VariablesContainer()
+        let container2 = VariablesContainer()
+
+        container1.addObjectList(list, for: objectA)
+
+        container2.addObjectList(list, for: objectA)
+        container2.addObjectList(list, for: objectB)
+
+        XCTAssertFalse(container1.isEqual(to: container2))
+    }
+
+    func testIsEqualToVariablesContainerForSameNameDifferentObject() {
+        let objectA = SpriteObject()
+        objectA.name = "testObjectA"
+
+        let objectB = SpriteObject()
+        objectB.name = "testObjectA"
+
+        let list = UserVariable(name: "testList", isList: true)
+
+        let container1 = VariablesContainer()
+        let container2 = VariablesContainer()
+
+        container1.addObjectList(list, for: objectA)
+        container2.addObjectList(list, for: objectB)
+
+        XCTAssertTrue(container1.isEqual(to: container2))
+    }
+
+    func testIsEqualToVariablesContainerForVariableWithSameNameDifferentValues() {
+        let objectA = SpriteObject()
+        objectA.name = "testObjectA"
+
+        let variable1 = UserVariable(name: "testvariable", isList: false)
+        variable1?.value = 10
+
+        let variable2 = UserVariable(name: "testvariable", isList: false)
+        variable2?.value = 20
+
+        let container1 = VariablesContainer()
+        let container2 = VariablesContainer()
+
+        container1.addObjectVariable(variable1, for: objectA)
+        container2.addObjectVariable(variable2, for: objectA)
+
+        XCTAssertFalse(container1.isEqual(to: container2))
+    }
+
+    func testIsEqualToVariablesContainerForItemWithSameNameDifferentType() {
+        let objectA = SpriteObject()
+        objectA.name = "testObjectA"
+
+        let list = UserVariable(name: "testUserVariable", isList: true)
+        let variable = UserVariable(name: "testUserVariable", isList: false)
+
+        let container1 = VariablesContainer()
+        let container2 = VariablesContainer()
+
+        container1.addObjectList(list, for: objectA)
+        container2.addObjectVariable(variable, for: objectA)
+
+        XCTAssertFalse(container1.isEqual(to: container2))
+    }
+
+    func testMutableCopy() {
+        let objectA = SpriteObject()
+        objectA.name = "testObjectA"
+
+        let list = UserVariable(name: "testList", isList: true)
+        let variable = UserVariable(name: "testvariable", isList: false)
+
+        let container = VariablesContainer()
+
+        container.addObjectList(list, for: objectA)
+        container.addObjectVariable(variable, for: objectA)
+        container.setUserVariable(variable, toValue: 10)
+
+        let copyContainer = container.mutableCopy() as! VariablesContainer
+
+        XCTAssertTrue(container.isEqual(to: copyContainer))
+        XCTAssertFalse(container == copyContainer)
+
+        container.change(variable, byValue: 10)
```

Comment:
Does that implicitly mean that there is no deep copy of the items of the container (i.e. `list` and `variable`)? If so, please better remove the variable change and replace it with pseudo code: `container.list has same reference than copyContainer.list `(and the same with `variable`).

Reply:
Yes, there is no deep copy of the items of the container.
sorry I couldn't understand, are you asking to write the pseudo-code: `container.list has same reference than copyContainer.list` in comment which replaces `container.change(variable, byValue: 10)`

Reply:
Yes please replace this, but not with a comment, but with actual code (i.e. `XCTAssertTrue(container.allVariables()[0] === copyContainer.allVariables()[0])` or similar).

Reply:
Sorry, I could understand it😛, I will make the changes.